### PR TITLE
DR2 1569 migrate to scala 3

### PR DESCRIPTION
**Pull request by MancunianSam**
- **Move to Scala3**
- **Update scalac options**
- **Ran scalafmt**
- **Update dependencies and replace last implicit**
- **Update dependencies and replace last implicit**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -20,7 +18,7 @@ lazy val commonSettings = Seq(
     mockito % Test,
     scalaTest % Test
   ),
-  scalaVersion := "2.13.13",
+  scalaVersion := "3.3.1",
```
Comment (techncl) [inline build.sbt:21]: The latest Scala 3 LTS version is now 3.3.3

File: project/Dependencies.scala [hunk 2]
```diff
@@ -12,10 +13,10 @@ object Dependencies {
   lazy val circe = "io.circe" %% "circe-core" % circeVersion
   lazy val dynamoDB = "software.amazon.awssdk" % "dynamodb" % awsSdkVersion
   lazy val eventBridgeSdk = "software.amazon.awssdk" % "eventbridge" % awsSdkVersion
-  lazy val mockito = "org.mockito" %% "mockito-scala" % "1.17.31"
+  lazy val mockito = "org.scalatestplus" %% "mockito-5-10" % s"$scalaTestVersion.0"
```
Comment (techncl) [inline project/Dependencies.scala:16]: How come there is a `.0` at the end?

Comment (MancunianSam) [reply project/Dependencies.scala:16]: It tracks the scalatest versions I think but it's published with this zero on the end. I'm not sure why.

**Comment by MancunianSam** [comment]
I can't see any reason why I removed the lazy keyword so I've put them back.

**Pull request merged**